test/powershell/Modules/Microsoft.PowerShell.Utility/WebCmdlets.Tests.ps1

```diff
@@ -296,6 +296,46 @@ function ExecuteWebRequest
     return $result
 }
 
+# Starts the ClientCertificateCheck listener
+function StartClientCertificateCheck 
+{
+    param([int]$Port)
+    $initTimeoutSeconds  = 15
+    $projectRootPath     = Resolve-Path (Join-Path $PSScriptRoot "\..\..\..\..\")
+    $appSrcPath          = Join-Path $projectRootPath  'test\tools\ClientCertificateCheck'
+    $appPublishPath      = Join-Path $appSrcPath 'bin'
+    $appDllPath          = Join-Path $appPublishPath 'ClientCertificateCheck.dll'
+    $serverPfxPath       = Join-Path $appSrcPath 'ServerCert.pfx'
+    $serverPfxPassword   = 'password'
+    $initCompleteMessage = 'Now listening on'
+    
+    $timeOut = (get-date).AddSeconds($initTimeoutSeconds)
+    $Job = Start-Job {
```

Comment:
@TravisEz13 and @iSazonov Ok, since it appears we may head in the direction of using this for more tests than just the certificate authentication, I have renamed the app `HttpsListener` and added an `HttpsListener` module to house the `Start-`, `Stop`, and `Get-` commands. There is a module script variable to track the running listeners so if it is already listening on the requested port the running listener is returned. To keep it always running, I have removed the stop commands form the `AfterAll` blocks. I think that incorporates everything discussed here except making it a binary module.